homework/index.js

```diff
@@ -27,21 +12,151 @@
+      class: 'contributor-header',
+      text: 'Contributors',
+    });
+    const ul = createAndAppend('ul', rightContainer, {
+      class: 'contributor-list',
+    });
+    contributors.forEach(contributor => {
+      const li = createAndAppend('li', ul, {
+        class: 'contributor-container',
+      });
+      createAndAppend('img', li, {
+        class: 'contributor-avatar',
+        src: contributor.avatar_url,
+      });
+      const contributorDataDiv = createAndAppend('div', li, {
+        class: 'contributor-data',
+      });
+      createAndAppend('a', contributorDataDiv, {
+        text: contributor.login,
+        href: contributor.html_url,
+        class: 'contributor-name',
+      });
+      createAndAppend('div', contributorDataDiv, {
+        text: contributor.contributions,
+        class: 'contributionCount',
+      });
     });
   }
 
+  async function fetchContributors(url, rightContainer) {
+    try {
+      const fetchedContributors = await fetchJSON(url);
+      renderRepoContributors(fetchedContributors, rightContainer);
+    } catch (error) {
+      createAndAppend('div', rightContainer, {
+        text: `We got an error: $ {error.message}`,
+        class: 'alert-error',
+      });
+    }
+  }
+
+  function renderRepoInfoAndContributorsOnstartup(repositories, root) {
+    createAndAppend('img', root, {
+      src: './hyf.png',
+      id: 'hyf-logo',
+      alt: 'logo image',
+    });
+    const header = createAndAppend('header', root, {
+      class: 'header',
+    });
+    createAndAppend('p', header, {
+      text: 'HYF Repositories',
+    });
+    const selectMenu = createAndAppend('select', header, {
+      id: 'selectMenu',
+    });
+    const container = createAndAppend('div', root, {
+      id: 'container',
+    });
+    const leftContainer = createAndAppend('div', container, {
+      class: 'leftContainer',
+    });
+
+    const rightContainer = createAndAppend('div', container, {
+      class: 'rightContainer',
+    });
+
+    repositories.sort((a, b) => a.name.localeCompare(b.name));
+
+    repositories.forEach((repository, i) => {
+      createAndAppend('option', selectMenu, {
+        value: i,
+        text: repository.name,
+      });
+    });
+    const firstRepository = repositories[0];
+    renderRepositoryInfo(firstRepository, leftContainer);
+    fetchContributors(firstRepository.contributors_url, rightContainer);
+
+    selectMenu.addEventListener('change', event => {
+      leftContainer.innerHTML = '';
+      renderRepositoryInfo(repositories[event.target.value], leftContainer);
+      rightContainer.innerHTML = '';
+      fetchContributors(repositories[event.target.value].contributors_url, rightContainer);
+    });
+  }
+
+  function main(url) {
+    async function fetchAllData() {
+      const root = document.getElementById('root');
+      const parsedResponse = await fetchJSON(url);
+      renderRepoInfoAndContributorsOnstartup(parsedResponse, root);
+    }
+    fetchAllData();
```

Comment:
A function as you know made to be able re-use, So what about passing the url from here? and moving the function outside of the main making it more specific with it's doings ?